src/katgpucbf/xbgpu/main.py

```diff
@@ -217,6 +276,17 @@ def parse_args(arglist: Sequence[str] | None = None) -> argparse.Namespace:
     if args.sample_bits != 8:
         parser.error("Only 8-bit values are currently supported.")
 
+    # Convert from *BOutputDict to *BOutput
+    used_names = {"xengine"}
+    for output in args.beam:
+        name = output["name"]
+        if name in used_names:
+            parser.error(f"output name {name} already used.")
+        used_names.add(name)
+        if "channels_per_substream" not in output:
+            output["channels_per_substream"] = args.channels_per_substream
```

Comment:
I am currently leaning towards suggesting removing this.

It shouldn't be possible for the beamformer to have a different number of channels per substream than the X-engine that it's co-located with.

Reply:
In theory the output could be subdivided more finely than the input, and maybe someone will want that (e.g. to parallelise a beam across more receivers). But I'm not expecting that to happen, and we can add CLI support for that later if we decide to implement it. For now I'd recommend that channels-per-substream is considered an input property, not an output property.